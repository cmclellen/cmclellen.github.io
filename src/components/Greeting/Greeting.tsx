import  { FC } from 'react';
import { GreetingWrapper } from './Greeting.styled';

interface GreetingProps {
   message?: string;
}

const Greeting: FC<GreetingProps> = (props: GreetingProps) => {

   const { message= 'World' } = props

   return (
      <GreetingWrapper data-testid="Greeting">
         <div>Hello, {message}!</div>
      </GreetingWrapper>
      );
   }

export default Greeting;
