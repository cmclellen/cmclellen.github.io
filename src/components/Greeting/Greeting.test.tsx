import renderer from 'react-test-renderer'
import { describe, expect, it} from 'vitest'
import Greeting from './Greeting';

describe('<Greeting />', () => {
  it('Greeting component renders correctly', () => {
    const component = renderer.create(
      <Greeting/>
    )

    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('prop working', () => {
    const component = renderer.create(
      <Greeting message={'Universe'} />
    )

    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
});