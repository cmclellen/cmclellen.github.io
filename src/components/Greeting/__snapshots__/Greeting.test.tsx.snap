// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Greeting /> > Greeting component renders correctly 1`] = `
<div
  className="sc-bdvuGq hka-DTH"
  data-testid="Greeting"
>
  <div>
    Hello, 
    World
    !
  </div>
</div>
`;

exports[`<Greeting /> > prop working 1`] = `
<div
  className="sc-bdvuGq hka-DTH"
  data-testid="Greeting"
>
  <div>
    Hello, 
    Universe
    !
  </div>
</div>
`;
